### PR TITLE
Minor refinements

### PR DESCRIPTION
**Pull request by daniel-hall**
- Removed Source(wrappedValue:) initializer to encourage injection and get correct memberwise initializer parameter
- Made ActionExecution the output for the ActionExecution.errors Publisher to send along full information
- Added a Error? property to ActionExecution
- Action’s callAsFunction method’s ifUnavailable closure we updated to return a Bool that indicates whether the error should also propagate to the ActionExecution.errors publisher or not
- Updated to have FilePersistence and UserDefaultsPersistence APIs and their respective Descriptor match each other and match CachePersistence’s API and Descriptor
- Made Persistable<Value>.Found’s isExpired property a Bool instead of a closure that returns a Bool to simplify, since a new value is sent on expiration anyway
- Updated `ifUnavailable` closures when invoking Actions to receive the full ActionExecution instance in order to capture the input as well as any error
- Made Source<Model>’s subscribe method return a discardable return type of the instance of the Source itself. This allows assigning and subscribing in a single expression, e.g. `_source = source.subscribing(self, method: method)`
- Made small cleanups to sample code syntax
- Implemented workaround for iOS 16 SwiftUI scrollTo() crash in ToDoList example apps
- Added some doc comments for DataConvertible, Connectable, ConnectableWithPlaceholder, FileDescriptor and UserDefaultsDescriptor

### CODE REVIEW

File: Sources/SourceArchitecture/CommonPersistence/FilePersistence.swift [hunk 1]
```diff
@@ -29,65 +29,90 @@ import Foundation
...
 
     public init() { }
 
-    public subscript<Value>(descriptor: Descriptor<Value>) -> Source<Persistable<Value>> {
+    public func persistableSource<Value>(for descriptor: FileDescriptor<Value>) -> Source<Persistable<Value>> {
         dictionary[descriptor.url.absoluteString] { FilePersistenceSource(descriptor: descriptor).eraseToSource() }
     }
 }
 
+public struct FileDescriptor<Value> {
```
Comment (albertbori) [inline Sources/SourceArchitecture/CommonPersistence/FilePersistence.swift:41]: nit: Some `///` comments on this type and its public members describing what they are and how to use them would be helpful.

File: Sources/SourceArchitecture/CommonPersistence/UserDefaultsPersistence.swift [hunk 2]
```diff
@@ -29,30 +29,54 @@ import Foundation
...
 
     public init() { }
 
-    public subscript<Value>(descriptor: Descriptor<Value>) -> Source<Persistable<Value>> {
+    public func persistableSource<Value>(for descriptor: UserDefaultsDescriptor<Value>) -> Source<Persistable<Value>> {
         dictionary[descriptor.key] { UserDefaultsSource(descriptor: descriptor).eraseToSource() }
     }
 }
 
+public struct UserDefaultsDescriptor<Value> {
```
Comment (albertbori) [inline Sources/SourceArchitecture/CommonPersistence/UserDefaultsPersistence.swift:41]: nit: Some `///` comments on this type and its public members describing what they are and how to use them would be helpful.

File: Sources/SourceArchitecture/CoreModels/Connectable.swift [hunk 3]
```diff
@@ -75,11 +75,11 @@ public enum Connectable<Value> {
         connected != nil
     }
     
-    public func connect(ifUnavailable: ((Swift.Error) -> Void)? = nil) {
+    public func connect(ifUnavailable: ((Swift.Error) -> Bool)? = nil) {
```
Comment (albertbori) [inline Sources/SourceArchitecture/CoreModels/Connectable.swift:78]: nit: Some `///` comments on these functions would be helpful, especially in clarifying the meaning of the `Bool` return value.

File: Sources/SourceArchitecture/CoreModels/ConnectableWithPlaceholder.swift [hunk 4]
```diff
@@ -86,11 +86,11 @@ public enum ConnectableWithPlaceholder<Value, Placeholder> {
         }
     }
 
-    public func connect(ifUnavailable: ((Swift.Error) -> Void)? = nil) {
+    public func connect(ifUnavailable: ((Swift.Error) -> Bool)? = nil) {
```
Comment (albertbori) [inline Sources/SourceArchitecture/CoreModels/ConnectableWithPlaceholder.swift:89]: nit: Same here

File: Sources/SourceArchitecture/CommonPersistence/DataConvertible.swift [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//
+
+import Foundation
+
+public protocol DataConvertible {
```
Comment (albertbori) [inline Sources/SourceArchitecture/CommonPersistence/DataConvertible.swift:28]: nit: Some `///` comments on this type and its public members describing what they are and how to use them would be helpful.

**Review by albertbori — APPROVED** [review]
Looks great! Left a couple of nits.

**Comment by daniel-hall** [comment]
Thanks for keeping me honest, @albertbori!  😁

I added the doc comments you suggested!

**Pull request merged**